src/save_load_scfres.jl

```diff
@@ -0,0 +1,42 @@
+"""
+    load_scfres(filename)
+
+Load back an `scfres`, which has previously been stored with `save_scfres`.
+Note the warning in `save_scfres`.
+"""
+function load_scfres end
+
+
+
+"""
+    save_scfres(filename, scfres)
+
+Save an `scfres` obtained from `self_consistent_field` to a JLD2 or VTK file depending on the extension.
+
+!!! warning "No compatibility guarantees"
+    No guarantees are made with respect to this function at this point.
+    It may change incompatibly between DFTK versions or stop working / be removed
+    in the future.
+"""
+function save_scfres(filename::AbstractString, scfres::NamedTuple)
+    fs = split(filename,".")
+    @assert length(fs) != 1 "No file extension mentioned"
+    
+    if(fs[end] == "vts")
```

Comment:
I would use `Val(Symbol(fs[end]))` directly to dispatch. If you then have a generic `save_scfres(filename, scfres, format)` you can use that to emit error messages.

Reply:
I have not defined the generic three argument function. So I am creating the three argument function in the `save_load_scfres.jl`, where I am dumping the error methods.

Reply:
Sure, that's fine. You can just put the generic fallback to `save_load_scfres.jl` as well.

Reply:
I am curious about this decision. If both ways give us the same results(here errors) then why can do we care about another function dispatch? Is there some special advantage of dispatch here or is it just  a design thing?

Reply:
Because it avoids explicitly writing the `if`s, which saves you about 10 lines of code, I'd say ... so yes, it's a design thing.